polynote-frontend/polynote/ui/component/toolbar.ts

```diff
@@ -36,28 +36,30 @@ export class Toolbar extends Disposable {
                 const nbInfo = ServerStateHandler.getOrCreateNotebook(path);
                 if (nbInfo?.info) {
                     currentNotebookHandler = nbInfo.handler
-                    const newListener = currentNotebookHandler.addObserver(state => {
-                        if (state.activeCellId !== undefined) {
-                            const lang = state.cells[state.activeCellId].language
+                    const newListener = currentNotebookHandler.activeCellView.addObserver(activeCell => {
+                        if (activeCell !== undefined) {
+                            const lang = activeCell.language
                             if (lang === "text") {
                                 this.el.classList.remove('editing-code');
                                 this.el.classList.add('editing-text');
                             } else {
                                 this.el.classList.remove('editing-text');
                                 this.el.classList.add('editing-code');
                             }
+                        } else {
+                            this.el.classList.remove('editing-code');
+                            this.el.classList.remove('editing-text');
                         }
                     }).disposeWith(this);
-                    if (currentNotebookHandler) {
-                        if (cellSelectionListener !== undefined)
-                            cellSelectionListener.tryDispose();
-                        cellSelectionListener = newListener;
-                        currentNotebookHandler = nbInfo.handler;
-                        nb.enable(nbInfo.info.dispatcher);
-                        cell.enable(currentNotebookHandler);
-                        code.enable(nbInfo.info.dispatcher);
-                        text.enable();
-                    }
+
+                    if (cellSelectionListener !== undefined)
+                        cellSelectionListener.tryDispose();
+                    cellSelectionListener = newListener;
+                    currentNotebookHandler = nbInfo.handler;
```

Comment:
Nice catch finding that `if (currentNotebookHandler)` was unnecessary here. Likewise, I think this line is redundant too because we're already setting the same thing in line 38.